docs/api-reference/mapbox/overview.md

```diff
@@ -37,156 +27,38 @@ npm install @deck.gl/mapbox
 
-```js
-import mapboxgl from 'mapbox-gl';
-
-mapboxgl.accessToken = '<your access token here>';
-const map = new mapboxgl.Map({
-  container: 'map',
-  style: 'mapbox://styles/mapbox/streets-v9',
-  center: [-74.50, 40],
-  zoom: 9,
-  antialias: true // Mapbox disables WebGL's antialiasing by default
-});
-
-map.on('load', () => {
-  map.addLayer(myDeckLayer);
-});
-```
+One major use case for interleaving deck.gl and Mapbox is that some important information in the Mapbox map could be hidden by a deck.gl visualization layer, and controlling opacity is not enough. A typical example of this is labels and roads, where it is desirable to have a deck.gl visualization layer render on top of the Mapbox geography, but where one might still want to see e.g. labels and/or roads. Alternatively, the deck.gl visualization should cover the ground, but not the roads and labels.
 
+To inject a deck layer into the Mapbox stack, either:
 
-### Using with React
+- Create a [MapboxLayer](/docs/api-reference/mapbox/mapbox-layer.md) and call the [`map.addLayer(layer, before?)`](https://www.mapbox.com/mapbox-gl-js/api/#map#addlayer) API.
+- Add a `beforeId` prop to any layer passed to the [MapboxOverlay](/docs/api-reference/mapbox/mapbox-overlay.md) control.
 
-```js
-import React, {useState, useRef, useCallback} from 'react';
-import DeckGL from '@deck.gl/react';
-import {ScatterplotLayer} from '@deck.gl/layers';
-import {StaticMap} from 'react-map-gl';
-
-import {MapboxLayer} from '@deck.gl/mapbox';
-
-const INITIAL_VIEW_STATE = {
-  longitude: -74.50,
-  latitude: 40,
-  zoom: 9
-};
-
-const data = [
-  {position: [-74.5, 40], size: 100}
-];
-
-function App() {
-  // DeckGL and mapbox will both draw into this WebGL context
-  const [glContext, setGLContext] = useState();
-  const deckRef = useRef(null);
-  const mapRef = useRef(null);
-
-  const onMapLoad = useCallback(() => {
-    const map = mapRef.current.getMap();
-    const deck = deckRef.current.deck;
-
-    // You must initialize an empty deck.gl layer to prevent flashing
-    map.addLayer(
-      // This id has to match the id of the deck.gl layer
-      new MapboxLayer({ id: "my-scatterplot", deck }),
-      // Optionally define id from Mapbox layer stack under which to add deck layer
-      // 'before-layer-id'
-    );
-  }, []);
-
-  const layers = [
-    new ScatterplotLayer({
-      id: 'my-scatterplot',
-      data,
-      getPosition: d => d.position,
-      getRadius: d => d.size,
-      getFillColor: [255, 0, 0]
-    })
-  ];
-
-  return (
-    <DeckGL
-      ref={deckRef}
-      layers={layers}
-      initialViewState={INITIAL_VIEW_STATE}
-      controller={true}
-      onWebGLInitialized={setGLContext}
-      glOptions={{
-        /* To render vector tile polygons correctly */
-        stencil: true
-      }}
-    >
-      {glContext && (
-        /* This is important: Mapbox must be instantiated after the WebGLContext is available */
-        <StaticMap
```

Comment:
How can we use react-map-gl and interleave mapbox?

Reply:
Added an example to `mapbox-overlay.md`, basically [this](https://github.com/visgl/react-map-gl/pull/1887)